service/matching/matchingEngine.go

```diff
@@ -622,8 +622,8 @@ func (e *matchingEngineImpl) createPollForDecisionTaskResponse(context *taskCont
 		}
 		token, _ = e.tokenSerializer.Serialize(taskoken)
 		if context.syncResponseCh == nil {
-			scope := domainTaggedMetricScope(e.domainCache, task.DomainID, e.metricsClient, metrics.MatchingPollForDecisionTaskScope)
-			scope.RecordTimer(metrics.AsyncMatchLatency, time.Since(task.CreatedTime))
+			scope := e.metricsClient.Scope(metrics.MatchingPollForDecisionTaskScope)
+			scope.Tagged(metrics.DomainTag(context.domainName)).RecordTimer(metrics.AsyncMatchLatency, time.Since(task.CreatedTime))
```

Comment:
does scope.Tagged() deal with empty string as parameter correctly ?

Reply:
I don't know but it's always better to have a checker. Will add it here.